### PR TITLE
Docs/DOC-1068 Schema updates

### PR DESCRIPTION
**Pull request by NotRobSheridan**
#### Description
Updated docs to reflect new schema generation workflow. 

Pages edited: 
schema.md 
gdk-template.md
tutorial-multiserver-setup.md 
tutorial-porting-guide.md

Also updated toolbar screenshots. 

#### Tests
Local improbadoc

#### Primary reviewers
@ernopp @m-samiec @ElleEss

### CODE REVIEW

**Comment by NotRobSheridan** [comment]
schema-button-full-scan
snapshot-button
start-button
stop-button
toolbar-buttons
inspector-button
play-button
schema-button

File: SpatialGDK/Documentation/content/spatialos-concepts/schema.md [hunk 1]
```diff
@@ -7,7 +7,17 @@ Schema is defined in `.schema` files and written in schemalang.  When you use th
 
 #### How to generate schema
 
-To generate schema, select the **Schema** button in the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons). The GDK automatically iterates through classes with replicated properties to generate the required schema files and then updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase).
+**Generating Schema for the first time**
+
+In the Unreal Editor, on the GDK toolbar, open the **Schema** drop-down menu and select **Schema (Full Scan)**. <br/>You must run a full scan the first time you generate Schema for a project. When you select Full Scan, the GDK automatically iterates through all classes with replicated properties to generate the required schema files and then updates the [SchemaDatabase](http://localhost:8080/reference/1.0/content/glossary#schemadatabase). <br/>
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/spatialos-concepts/schema.md]: Nits: 
Do we need to say "required schema files" it reads a bit awkwardly?
My pedantry wants to make it "when you select **Schema (Full Scan)**".

```suggestion
In the Unreal Editor, on the GDK toolbar, open the **Schema** drop-down menu and select **Schema (Full Scan)**. <br/>You must run a full scan the first time you generate Schema for a project. When you select **Schema (Full Scan)**, the GDK automatically iterates through all classes with replicated properties to generate the schema files and then updates the [SchemaDatabase](http://localhost:8080/reference/1.0/content/glossary#schemadatabase). <br/>
```

File: SpatialGDK/Documentation/content/spatialos-concepts/schema.md [hunk 2]
```diff
@@ -7,7 +7,17 @@ Schema is defined in `.schema` files and written in schemalang.  When you use th
...
+
+![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/schema-button-full-scan.png)
+
+_Image: In the GDK toolbar in the Unreal Editor, select **Schema (Full Scan)**_
+
+**Generating Schema after running a full scan**
+
+To generate schema after you have run a full scan, select  **Schema** in the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons). The GDK automatically iterates through classes with replicated properties that aren't already referenced in the SchemaDatabase. It then automatically generates the required schema files and updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase).
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/spatialos-concepts/schema.md]:
```suggestion
To generate schema after you have run a full scan, select  **Schema** in the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons). The GDK automatically iterates through classes with replicated properties that aren't already referenced in the SchemaDatabase. It then automatically generates the schema files and updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase).
```

File: SpatialGDK/Documentation/content/spatialos-concepts/schema.md [hunk 3]
```diff
@@ -7,7 +7,17 @@ Schema is defined in `.schema` files and written in schemalang.  When you use th
...
+
+In the Unreal Editor, on the GDK toolbar, open the **Schema** drop-down menu and select **Schema (Full Scan)**. <br/>You must run a full scan the first time you generate Schema for a project. When you select **Schema (Full Scan)**, the GDK automatically iterates through all classes with replicated properties to generate the schema files and then updates the [SchemaDatabase](http://localhost:8080/reference/1.0/content/glossary#schemadatabase). <br/>
+
+![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/schema-button-full-scan.png)
+
+_Image: In the GDK toolbar in the Unreal Editor, select **Schema (Full Scan)**_
+
+**Generating Schema after running a full scan**
+
+To generate schema after you have run a full scan, select  **Schema** in the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons). The GDK automatically iterates through classes with replicated properties that aren't already referenced in the SchemaDatabase. It then automatically generates the schema files and updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase).
```
Comment (m-samiec) [inline SpatialGDK/Documentation/content/spatialos-concepts/schema.md]: This isn't quite true. The iterative scan analyses all objects currently in memory (currently loaded by the editor), and adds/updates schema definitions where necessary. If the asset isn't in memory though, it won't be added to the SchemaDatabase.

Comment (m-samiec) [reply SpatialGDK/Documentation/content/spatialos-concepts/schema.md]: The *When to generate schema* section could be updated also.

File: SpatialGDK/Documentation/content/spatialos-concepts/schema.md [hunk 4]
```diff
@@ -7,25 +7,45 @@ Schema is defined in `.schema` files and written in schemalang.  When you use th
...
 #### How to generate schema
 
-To generate schema, select the **Schema** button in the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons). The GDK automatically iterates through classes with replicated properties to generate the required schema files and then updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase).
+There are two ways to generate schema for your project:
 
-![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/schema-button.png)
+* **Generate schema (Full Scan)** 
 
-_Image: In the GDK toolbar in the Unreal Editor, select **Schema**_
+    To generate schema for all objects with [replicated properties (Unreal documentation)](https://docs.unrealengine.com/en-US/Gameplay/Networking/Actors/Properties) in your project, in the Unreal Editor, on the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons), open the **Schema** drop-down menu and select **Schema (Full Scan)**.<br/> You must select **Schema (Full Scan)** the first time you generate schema for a project.
```
Comment (m-samiec) [inline SpatialGDK/Documentation/content/spatialos-concepts/schema.md]: We also generate schema for objects that have RPCs. Here's unreal documentation on them - https://docs.unrealengine.com/en-us/Gameplay/Networking/Actors/RPCs

File: SpatialGDK/Documentation/content/spatialos-concepts/schema.md [hunk 5]
```diff
@@ -7,25 +7,45 @@ Schema is defined in `.schema` files and written in schemalang.  When you use th
...
 As the GDK automatically generates all the schema you need, you do not have to write or edit schema manually when using the GDK.
 
 #### When to generate schema
 
-You must generate schema when you add or change any [replicated properties (Unreal documentation)](https://docs.unrealengine.com/en-US/Gameplay/Networking/Actors/Properties) that you want to deploy to SpatialOS.
+Select **Schema (Full Scan)** if you have: 
 
-The GDK only generates schema for classes currently loaded into memory. This means if your project uses [sublevels](<https://docs.unrealengine.com/en-us/Engine/Levels/LevelsWindow>), you’ll need to load them in addition to your map, before generating schema.
+* Not generated any schema for your project.
+* Added an object with replicated properties to a level that is not currently loaded by the Editor.
```
Comment (m-samiec) [inline SpatialGDK/Documentation/content/spatialos-concepts/schema.md]: References to `object` should probably be replaced with `class`.

File: SpatialGDK/Documentation/content/spatialos-concepts/schema.md (hunk 4 above)
Comment (m-samiec) [reply SpatialGDK/Documentation/content/spatialos-concepts/schema.md]: It's very much a nit, as 99% of the time any class that has an RPC will have replicated properties. You also don't need to regenerate schema when modifying RPCs, we just use the existence of RPCs to ensure the core data schema component is generated in the first place.

File: SpatialGDK/Documentation/content/spatialos-concepts/schema.md [hunk 6]
```diff
@@ -1,31 +1,56 @@
...
 #### How to generate schema
 
-To generate schema, select the **Schema** button in the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons). The GDK automatically iterates through classes with replicated properties to generate the required schema files and then updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase).
+There are two ways to generate schema for your project:
 
-![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/schema-button.png)
+* **Generate schema (Full Scan)** 
 
-_Image: In the GDK toolbar in the Unreal Editor, select **Schema**_
+    To generate schema for all classes* in your project:<br/>
```
Comment (m-samiec) [inline SpatialGDK/Documentation/content/spatialos-concepts/schema.md]:
```suggestion
    To generate schema for all classes in your project:<br/>
```

File: SpatialGDK/Documentation/content/spatialos-concepts/schema.md [hunk 7]
```diff
@@ -1,31 +1,56 @@
...
+
+    In the Unreal Editor, on the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons), open the **Schema** drop-down menu and select **Schema (Full Scan)**.<br/> You must select **Schema (Full Scan)** the first time you generate schema for a project. 
+    <br/> ![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/schema-button-full-scan.png)<br/>
+    _Image: In the GDK toolbar in the Unreal Editor, select **Schema (Full Scan)**_<br/>
+    <br/>When you select **Schema (Full Scan)**, the GDK automatically iterates through all classes* in your project to generate the schema files, and then updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase). <br/>
+
+    Run a full scan the first time you generate schema for your project, and whenever you need to generate schema for classes* that are not currently loaded by the Editor.<br/>For example: You need to select **Schema (Full Scan)** if you didn't generate schema after adding a Blueprint to a level, and that level is no longer open in the Editor.<br/><br/>
```
Comment (m-samiec) [inline SpatialGDK/Documentation/content/spatialos-concepts/schema.md]:
```suggestion
    Run a full scan the first time you generate schema for your project, and whenever you need to generate schema for classes* that are not currently loaded by the Editor.<br/>For example: You need to select **Schema (Full Scan)** if you didn't generate schema after adding a new Blueprint to your game, and that Blueprint is no longer open in the Editor.<br/><br/>
```

File: SpatialGDK/Documentation/content/spatialos-concepts/schema.md [hunk 8]
```diff
@@ -1,31 +1,56 @@
...
+
+* **Generate Schema (Iterative)**
+
+    To generate schema for classes* that are currently loaded by the Editor: <br/>
+
+    Select  **Schema** in the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons). The GDK automatically iterates through classes* that are currently loaded by the Editor, generates the schema files and updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase).<br/>
```
Comment (m-samiec) [inline SpatialGDK/Documentation/content/spatialos-concepts/schema.md]:
```suggestion
    Select **Schema** in the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons). The GDK automatically iterates through classes* that are currently loaded by the Editor, generates the schema files and updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase).<br/>
```

File: SpatialGDK/Documentation/content/spatialos-concepts/schema.md [hunk 9]
```diff
@@ -1,31 +1,56 @@
...
+    ![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/schema-button.png)<br/>
+    _Image: In the GDK toolbar in the Unreal Editor, select **Schema**_<br/><br/>
+
+    You must run an iterative scan when you add or change any classes* that you want to deploy to SpatialOS.
```
Comment (m-samiec) [inline SpatialGDK/Documentation/content/spatialos-concepts/schema.md]:
```suggestion
    You must run an iterative scan when you add or change replicated properties in any classes* that you want to deploy to SpatialOS.
```

File: SpatialGDK/Documentation/content/spatialos-concepts/schema.md [hunk 10]
```diff
@@ -1,31 +1,56 @@
...
 
+> \* Whenever you generate schema, the GDK automatically creates schema for classes with [replicated properties (Unreal documentation)](https://docs.unrealengine.com/en-US/Gameplay/Networking/Actors/Properties) or [RPCs (Unreal documentation)](https://docs.unrealengine.com/en-us/Gameplay/Networking/Actors/RPCs). If a class does not have replicated properties or RPCs, then the GDK does not generate schema for it. 
+
 #### When to generate schema
 
-You must generate schema when you add or change any [replicated properties (Unreal documentation)](https://docs.unrealengine.com/en-US/Gameplay/Networking/Actors/Properties) that you want to deploy to SpatialOS.
+Select **Schema (Full Scan)** if you have: 
+
+* Not generated any schema for your project.
+* Added a class with replicated properties to a level that is not currently loaded by the Editor.
```
Comment (m-samiec) [inline SpatialGDK/Documentation/content/spatialos-concepts/schema.md]:
```suggestion
* Added a class with replicated properties that is not currently loaded by the Editor.
```

File: SpatialGDK/Documentation/content/spatialos-concepts/schema.md [hunk 11]
```diff
@@ -1,31 +1,56 @@
...
+* Edited a class with replicated properties that is not currently loaded by the Editor.
 
-The GDK only generates schema for classes currently loaded into memory. This means if your project uses [sublevels](<https://docs.unrealengine.com/en-us/Engine/Levels/LevelsWindow>), you’ll need to load them in addition to your map, before generating schema.
+Select **Schema** if you have:
+
+* Added a class with replicated properties to a level that is currently loaded by the Editor.
```
Comment (m-samiec) [inline SpatialGDK/Documentation/content/spatialos-concepts/schema.md]:
```suggestion
* Added a class with replicated properties that is currently loaded by the Editor.
```

File: SpatialGDK/Documentation/content/spatialos-concepts/schema.md (hunk 9 above)
Comment (m-samiec) [reply SpatialGDK/Documentation/content/spatialos-concepts/schema.md]: Do we need this sentence? Perhaps just refer to the *When to generate schema* section

Comment (NotRobSheridan) [reply SpatialGDK/Documentation/content/spatialos-concepts/schema.md]: Nah, probably not. Good shout.

**Pull request merged**